### PR TITLE
* Removing implicit cast to prevent cast exceptions.

### PR DESCRIPTION
**Pull request by pranavkm**
- RemoteDependencyWalker null refs when match is non-null but the library
- is
- Print the package path correctly.

[Review on Reviewable](https://reviewable.io/reviews/nuget/nuget3/27)

### CODE REVIEW

File: src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs [hunk 1]
```diff
@@ -331,7 +331,7 @@ public Task<GraphNode<RemoteResolveResult>> WalkAsync(LibraryRange library, NuGe
                 // If we found an exact match then use it.
                 // This allows us to shortcircuit slow feeds even if there's an exact match
                 if (!libraryRange.VersionRange.IsFloating &&
-                    match != null &&
+                    match?.Library != null &&
```
Comment (pranavkm) [inline src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs]: Ran into this when trying it out locally. We have a match but it doesn't have a library. Is that a bug in the invoking code? cc @davidfowl

**Comment by pranavkm** [comment]
cc @anurse

File: src/NuGet.CommandLine/Program.cs [hunk 2]
```diff
@@ -75,7 +75,7 @@ public int Main(string[] args)
                             var packagesDir = packagesDirectory.HasValue() ?
                                 packagesDirectory.Value() :
                                 Path.Combine(Environment.GetEnvironmentVariable("USERPROFILE"), ".dnx", "packages");
-                            _log.LogVerbose($"Using packages directory: {packagesDirectory}");
+                            _log.LogVerbose($"Using packages directory: {packagesDir}");
```
Comment (yishaigalatzer) [inline src/NuGet.CommandLine/Program.cs:79]: format the file

File: src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs (hunk 1 above)
Comment (yishaigalatzer) [inline src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs]: Is this related to an unlisted package?

File: src/NuGet.ProjectModel/LockFileLibrary.cs [hunk 3]
```diff
@@ -50,13 +50,14 @@ public class LockFileTargetLibrary
     }
     public class LockFileItem
     {
-        public string Path { get; set; }
```
Comment (yishaigalatzer) [inline src/NuGet.ProjectModel/LockFileLibrary.cs:53]: one class per file

**Comment by yishaigalatzer** [comment]
:shipit: when comment are addressed and when you figured out the null Library thingie

File: src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs (hunk 1 above)
Comment (pranavkm) [inline src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs]: I didn't debug this - I ran into null refs here when trying to get the msbuild task running. @davidfowl did make this change recently so I wanted to check if he already knew of this.

Comment (davidfowl) [inline src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs]: Hmmm gotta check the dnu code, what provider matched?

**Comment by pranavkm** [comment]
<sup>**[src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs, line 334 [r1]](https://reviewable.io:443/reviews/nuget/nuget3/27#-JpHpXiGvrwMEnqPDTCn-r1-334)** ([raw file](https://github.com/nuget/nuget3/blob/b9434fba0f6000317305e21801d77c43a8b77ecc/src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs#L334)):</sup>
Nevermind - found the issue. I was returning a match when the library was null a few lines above this. This is fixed.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/nuget/nuget3/27)
